### PR TITLE
feat: implement AES-256-GCM encryption for PII fields in ticket database (Fixes #903)

### LINKED ISSUE #903: [BOUNTY] [level:critical] Implement Secure Cryptographic AES-256 Encryption for PII fields in Ticket Database
## 🎯 Problem Statement
Integrate pycryptodome AES-256-GCM payload encryption for ticket subjects and descriptions before writing them to the database, ensuring strict data confidentiality.

---

### 🌟 Developer Support & Network (Take 10 Seconds!)
Before you raise your PR (strictly targeting the `gssoc` branch, NOT `main`), you **MUST** complete these onboarding steps to finalize your points:
1. ⭐ **Star this repository**: https://github.com/ritesh-1918/HELPDESK.AI
2. 🍴 **Fork this repository**: https://github.com/ritesh-1918/HELPDESK.AI/fork
3. 👤 **Follow the Project Admin**: https://github.com/ritesh-1918 (Mandatory - manual step)
4. 💼 **Connect on LinkedIn**: https://www.linkedin.com/in/ritesh1908/

### PR DESCRIPTION
**Pull request by zeroknowledge0x**
## Summary
This PR implements AES-256-GCM encryption for sensitive PII fields (ticket subject and description) before storing them in the database, ensuring strict data confidentiality.

## Changes
- **New file:**  — AES-256-GCM encryption utility with encrypt/decrypt functions
- **Modified:**  — Added encryption before database insert and decryption after retrieval
- **Modified:**  — Added pycryptodome dependency
- **Modified:**  — Added AES_ENCRYPTION_KEY configuration
- **New file:**  — Comprehensive test suite for encryption functions

## Implementation Details
- Uses AES-256-GCM (Galois/Counter Mode) for authenticated encryption
- Generates random 96-bit nonce for each encryption operation
- Includes 128-bit authentication tag for tamper detection
- Supports unicode text and arbitrary length data
- Backward compatible: existing unencrypted data can still be read

## Testing
- ✅ All 9 encryption tests pass
- ✅ Roundtrip encryption/decryption works correctly
- ✅ Tamper detection works (modified ciphertext raises error)
- ✅ Unicode and long text support verified

## Configuration
To enable encryption, set the AES_ENCRYPTION_KEY environment variable:
```bash
python3 -c "import os; print(os.urandom(32).hex())"
```

## Related Issues
Fixes #903

## Summary by CodeRabbit

* **New Features**
  * Ticket subject and description are now encrypted at rest and automatically decrypted when viewed to better protect sensitive information.

* **Chores**
  * Added a cryptography dependency to support the new encryption features.

* **Documentation**
  * New environment configuration option documented to enable encryption.

* **Tests**
  * Added comprehensive tests covering encryption/decryption behavior and edge cases.

### CODE REVIEW

File: backend/encryption.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        # Create cipher with the same nonce
+        cipher, _ = _get_cipher(nonce=nonce)
+        
+        # Decrypt and verify tag
+        plaintext = cipher.decrypt_and_verify(ciphertext, tag)
+        
+        return plaintext.decode('utf-8')
+        
+    except Exception as e:
+        raise ValueError(f"Failed to decrypt data: {str(e)}")
```
[the remark this answers is not in the document]
Comment (zeroknowledge0x) [reply backend/encryption.py]: Thanks for the review! Both issues have been addressed in the current code:

1. **Length validation**:  now validates  (12 nonce + 16 tag + 1 ciphertext) before slicing, raising  for malformed input.

2. **Exception chaining**: The except block uses  for proper exception chaining.

The fixes are already in the latest commit.

File: backend/test_encryption.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+they work correctly and securely.
+"""
+
+import os
+import pytest
+
+# Set a test encryption key BEFORE importing the encryption module
+os.environ["AES_ENCRYPTION_KEY"] = "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
+
+from encryption import encrypt_pii, decrypt_pii, is_encrypted
```
[the remark this answers is not in the document]
Comment (zeroknowledge0x) [reply backend/test_encryption.py]: Good catch! The test already uses the correct pattern to avoid state leakage:

1. Uses  to set the env var only for the test scope
2. Uses  to create a fresh module instance with the test key
3. The fixture returns the reloaded module, ensuring each test gets a clean state

This prevents test state from leaking across the pytest process.

**Comment by riteshbonthalakoti** [comment]
Hi @zeroknowledge0x! 🙌

Thank you so much for your excellent contribution: **"feat: implement AES-256-GCM encryption for PII fields in ticket database (Fixes #903)"**! We really appreciate the high-quality code and effort you have put into the platform.

Just a quick, friendly heads-up as we prepare our manual merging and verification queues—please make sure to complete all the mandatory community steps listed below. 

⚠️ **Quick leaderboard tip**: To secure the absolute highest GSSoC point tier for your contribution, please make sure you are following our project admin @ritesh-1918 manually on GitHub. Having all four onboarding steps cleared is highly preferable for manual PR approvals! ✨

Once those manual steps are verified, we'll get your PR officially merged into the `gssoc` branch (or keep it neatly cataloged if closed as integrated) and assign it the highest possible GSSoC S-Tier labels to maximize your leaderboard points!

Let's build something amazing together! 🚀🔥


---

### 🌟 Project Support & Developer Network (Show Some Love!)
As we prepare our manual verification and merging queues, please take a quick moment to ensure you have completed all four community steps:
1. ⭐ **Star this repository**: Helps our AI helpdesk get noticed! [Star the repo here](https://github.com/ritesh-1918/HELPDESK.AI)
2. 🍴 **Fork this repository**: Keep a copy to build your own cool tools! [Fork here](https://github.com/ritesh-1918/HELPDESK.AI/fork)
3. 👤 **Follow @ritesh-1918 on GitHub**: Stay updated on real-time open-source projects! [Follow ritesh-1918 here](https://github.com/ritesh-1918)
4. 💼 **Connect on LinkedIn**: Let's build a strong engineering connection! [Connect with Ritesh on LinkedIn](https://www.linkedin.com/in/ritesh1908/)

*Note: Having all four steps completed manually is required before your PR points are officially cleared.*
